src/url-state-machine.js

```diff
@@ -93,6 +93,20 @@ function percentEncode(c) {
   return "%" + hex;
 }
 
+function percentDecode(input) {
```

Comment:
Oh, hmm, OK. Maybe using `byteLength` instead of `length` would clue me in on that, sorry.

It still seems a bit of a shame that most callers need to do all the jumping back and forth between byte sequences and strings, but I guess that's a spec inefficiency, and it's fine if we copy it.